File: app/src/main/java/io/github/freewebmovement/igniter/ui/component/textview/listener/RemotePort.java
package io.github.freewebmovement.igniter.ui.component.textview.listener;

import android.widget.TextView;

import io.github.freewebmovement.igniter.IgniterApplication;
import io.github.freewebmovement.igniter.persistence.TrojanConfig;

public class RemotePort extends TextViewListener {
    public RemotePort(TextView tv, IgniterApplication app) {
        super(tv, app);
    }

    protected void onTextChanged(String before, String old, String aNew, String after) {
        // update TextView
        if (tv.hasFocus()) {
            String portStr = tv.getText().toString();
            try {
                int port = Integer.parseInt(portStr);
                app.trojanConfig.setRemotePort(port);
            } catch (NumberFormatException e) {
                // Ignore when we get invalid number
                e.printStackTrace();
            }
        }
        endUpdates();
    }
}

File: app/src/main/java/io/github/freewebmovement/igniter/ui/component/textview/listener/RemotePort.kt
package io.github.freewebmovement.igniter.ui.component.textview.listener

import android.widget.TextView
import io.github.freewebmovement.igniter.IgniterApplication

class RemotePort(tv: TextView, app: IgniterApplication) : TextViewListener(tv, app) {
    override fun onTextChanged(before: String, old: String, aNew: String, after: String) {
        // update TextView
        if (tv.hasFocus()) {
            val portStr = tv.text.toString()
            try {
                val port = Integer.parseInt(portStr)
                app.trojanConfig.setRemotePort(port)
            } catch (e: NumberFormatException) {
                // Ignore when we get invalid number
                e.printStackTrace()
            }
        }
        endUpdates()
    }
}
